diff --git a/node_modules/@remix-run/dev/dist/vite/plugin.js b/node_modules/@remix-run/dev/dist/vite/plugin.js
index 134ad76..1bbba23 100644
--- a/node_modules/@remix-run/dev/dist/vite/plugin.js
+++ b/node_modules/@remix-run/dev/dist/vite/plugin.js
@@ -647,7 +647,7 @@ const remixVitePlugin = (remixUserConfig = {}) => {
           external: ssrExternals
         },
         optimizeDeps: {
-          entries: ctx.remixConfig.future.unstable_optimizeDeps ? [ctx.entryClientFilePath, ...Object.values(ctx.remixConfig.routes).map(route => path__namespace.join(ctx.remixConfig.appDirectory, route.file))] : [],
+          entries: ctx.remixConfig.future.unstable_optimizeDeps ? [importViteEsmSync.importViteEsmSync().normalizePath(ctx.entryClientFilePath),...Object.values(ctx.remixConfig.routes).map(route => resolveRelativeRouteFilePath(route, ctx.remixConfig))] : [],
           include: [
           // Pre-bundle React dependencies to avoid React duplicates,
           // even if React dependencies are not direct dependencies.
